hypervisor/dm/vpci/vpci.c

```diff
@@ -248,13 +248,11 @@ void init_vpci(struct acrn_vm *vm)
 	vpci_init_vdevs(vm);
 
 	vm_config = get_vm_config(vm->vm_id);
-	if (vm_config->load_order != PRE_LAUNCHED_VM) {
-		/* PCI MMCONFIG for post-launched VM is fixed to 0xE0000000 */
-		pci_mmcfg_base = (vm_config->load_order == SOS_VM) ? get_mmcfg_base() : 0xE0000000UL;
-		vm->vpci.pci_mmcfg_base = pci_mmcfg_base;
-		register_mmio_emulation_handler(vm, vpci_mmio_cfg_access,
+	/* PCI MMCONFIG for non-SOS is fixed to 0xE0000000 */
+	pci_mmcfg_base = (vm_config->load_order == SOS_VM) ? get_mmcfg_base() : 0xE0000000UL;
```

Comment:
Why we need to distinguish? The MACCROS are same too.

Also use DEFAULT_PCI_MMCFG_BASE?

Reply:
DEFAULT_PCI_MMCFG_BASE is the real physical address, post-launched VM is fixed to 0xE0000000, SO I think the pre-launched VM could use the virtual address too.

Reply:
Why we need fixed address?

Reply:
For post-launched VM, Device Model need to fill this fixed address in post-launched VM ACPI MCFG table.